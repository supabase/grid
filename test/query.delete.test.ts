import Query from '../src/query';

const query = new Query();

test('basic delete on table', async () => {
  const res = query
    .from('users')
    .delete()
    .filter('name', '=', 'hello world')
    .toSql();
  expect(res).toMatchSnapshot();
});
test('delete with returning record on table', async () => {
  const res = query
    .from('users')
    .delete({ returning: true })
    .filter('name', '=', 'hello world')
    .toSql();
  expect(res).toMatchSnapshot();
});
test('error on delete without filter', async () => {
  expect(() => {
    query.from('users').delete({ returning: true }).toSql();
  }).toThrow();
});
test('delete should ignore order', async () => {
  const res = query
    .from('users')
    .delete({ returning: true })
    .filter('name', '=', 'hello world')
    .order('name')
    .toSql();
  expect(res).toMatchSnapshot();
});
test('delete with comma symbol on filter value', async () => {
  const res = query
    .from('users')
    .delete()
    .filter('name', 'in', ['John, Bob and Billy', 'Marry, Jane and Fill'])
    .toSql();
  expect(res).toMatchSnapshot();
});
