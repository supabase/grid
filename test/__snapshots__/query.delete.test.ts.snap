// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic delete on table 1`] = `"delete from public.users where name = 'hello world';"`;

exports[`delete should ignore order 1`] = `"delete from public.users where name = 'hello world' returning *;"`;

exports[`delete with comma symbol on filter value 1`] = `"delete from public.users where name in ('John, Bob and Billy','Marry, Jane and Fill');"`;

exports[`delete with returning record on table 1`] = `"delete from public.users where name = 'hello world' returning *;"`;
